src/core/gui/dialog/SettingsDialog.cpp

```diff
@@ -907,16 +908,16 @@ void SettingsDialog::save() {
     settings->setStrokeRecognizerMinSize(
             static_cast<double>(gtk_spin_button_get_value(GTK_SPIN_BUTTON(get("spStrokeRecognizerMinSize")))));
 
-    int selectedInputDeviceIndex = gtk_combo_box_get_active(GTK_COMBO_BOX(get("cbAudioInputDevice"))) - 1;
-    if (selectedInputDeviceIndex >= 0 && selectedInputDeviceIndex < static_cast<int>(this->audioInputDevices.size())) {
-        settings->setAudioInputDevice(static_cast<int>(this->audioInputDevices[selectedInputDeviceIndex].getIndex()));
+    size_t selectedInputDeviceIndex =
+            static_cast<size_t>(gtk_combo_box_get_active(GTK_COMBO_BOX(get("cbAudioInputDevice")))) - 1;
+    if (selectedInputDeviceIndex < this->audioInputDevices.size()) {
+        settings->setAudioInputDevice(this->audioInputDevices[selectedInputDeviceIndex].getIndex());
     }
```

Comment:
This doesn't work (and didn't work) when switching from a non-default setting to the default setting. It should be possible to switch back to the system default.

Reply:
https://github.com/xournalpp/xournalpp/pull/4808/commits/3b3a1d58afbb3747e25e565d21fe07492c25192f should do the trick